dashboard/app/models/reference_guide.rb

```diff
@@ -18,4 +18,77 @@
 #  index_reference_guides_on_course_version_id_and_parent_key  (course_version_id,parent_reference_guide_key)
 #
 class ReferenceGuide < ApplicationRecord
+  include CurriculumHelper
+
+  belongs_to :course_version
+  validates_uniqueness_of :key, scope: :course_version_id
+  validate :key_format
+
+  def children
+    ReferenceGuide.where(course_version_id: course_version_id, parent_reference_guide_key: key)
+  end
+
+  def serialize
+    {
+      key: key,
+      course_version_key: course_version.key,
+      course_offering_key: course_version.course_offering.key,
+      parent_reference_guide_key: parent_reference_guide_key,
+      display_name: display_name,
+      content: content,
+      position: position
+    }
+  end
+
+  # writes reference guide to a seed file in the config directory
+  def write_serialization
+    return unless Rails.application.config.levelbuilder_mode
+    offering_and_version = "#{course_version.course_offering.key}-#{course_version.key}"
+    file_path = Rails.root.join("config/reference_guides/#{offering_and_version}/#{key}.json")
+    object_to_serialize = serialize
+    dirname = File.dirname(file_path)
+    unless File.directory?(dirname)
+      FileUtils.mkdir_p(dirname)
+    end
+    File.write(file_path, JSON.pretty_generate(object_to_serialize))
+  end
+
+  # runs through all seed files, creating and deleting records to match the seed files
+  def self.seed_all
+    # collect all existing ids
+    removed_records = all.pluck(:id)
+    Dir.glob(Rails.root.join("config/reference_guides/**/*.json")).each do |path|
```

Comment:
good choice on `.json`. I don't think there is a good reason for the `script_` part of `.script_json`.